recipes/tbb/all/conanfile.py

```diff
@@ -0,0 +1,153 @@
+                self.build_requires("make/4.2.1")
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.os == "Macos" and \
+           self.settings.compiler == "apple-clang" and \
+           Version(self.settings.compiler.version.value) < "8.0":
+            raise ConanInvalidConfiguration("%s %s couldn't be built by apple-clang < 8.0" % (self.name, self.version))
+        if self.settings.os != "Windows" and self.options.shared:
+            self.output.warn("Intel-TBB strongly discourages usage of static linkage")
+        if self.settings.os != "Windows" and self.options.tbbproxy and \
+           (not self.options.shared or \
+            not self.options.tbbmalloc):
+            raise ConanInvalidConfiguration("tbbproxy needs tbbmaloc and shared options")
+        if self.settings.os == "Windows" and self.options.tbbproxy and \
+           not self.options.tbbmalloc:
+            raise ConanInvalidConfiguration("tbbproxy needs tbbmaloc and shared options")
+        if self.settings.os == "Windows" and not self.options.shared:
+            raise ConanInvalidConfiguration("TBB could not be built as static lib on Windows")
+
+    @property
+    def is_msvc(self):
+        return self.settings.compiler == 'Visual Studio'
+
+    @property
+    def is_mingw(self):
+        return self.settings.os == 'Windows' and self.settings.compiler == 'gcc'
+
+    @property
+    def is_clanglc(self):
+        return self.settings.os == 'Windows' and self.settings.compiler == 'clang'
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("{}-{}".format(self.name.lower(), self.version.upper()), self._source_subfolder)
+
+        # Get the version of the current compiler instead of gcc
+        linux_include = os.path.join(self._source_subfolder, "build", "linux.inc")
+        tools.replace_in_file(linux_include, "shell gcc", "shell $(CC)")
+        tools.replace_in_file(linux_include, "= gcc", "= $(CC)")
+
+    def get_targets(self):
+        targets = ["tbb"]
+        if self.options.tbbmalloc:
+            targets.append("tbbmalloc")
+        if self.options.tbbproxy:
+            targets.append("tbbproxy")
+        return targets
+    
+    def build(self):
+        def add_flag(name, value):
+            if name in os.environ:
+                os.environ[name] += ' ' + value
+            else:
+                os.environ[name] = value
+
+        extra = "" if self.settings.os == "Windows" or self.options.shared else "extra_inc=big_iron.inc"
+        if self.settings.arch == "x86":
+            arch = "ia32" 
+        elif self.settings.arch == "x86_64":
+            arch = "intel64"
+        elif self.settings.arch == "armv7":
+            arch = "armv7"
+        elif self.settings.arch == "armv8":
+            arch = "aarch64"
```

Comment:
You could use a dict for this and it will be cleaner

Reply:
uses dict now